### ISSUE TITLE
readme: use authcallback, don't instantiate the lib with literal token

### ISSUE THREAD

**Issue by SimonWoolf** [ably/ably-dotnet#253]
from the readme:
```
var token = await client.Auth.RequestTokenAsync();
var tokenString = token.Token; // "xVLyHw.CLchevH3hF....MDh9ZC_Q"
var tokenClient = new AblyRest(new ClientOptions { TokenDetails = token });
```
people should never be instantiating the lib with just a token in production, so we shouldn't be encoruaging them to do it in the readme

**Comment by withakay** [comment]
@SimonWoolf are you suggesting we remove that section from the readme altogether or is there something that should live in its place?

**Comment by withakay** [comment]
@SimonWoolf as per out of band discussion, I think I should add a section similar to [this., from the iOS readme](https://github.com/ably/ably-ios#using-the-authcallback). WDYT?

**Comment by SimonWoolf** [comment]
@withakay Perfect :+1:

**Comment by withakay** [comment]
Fixed by #254
